File: src/UWP/ArcGISRuntime.UWP.Samples/Samples/Security/TokenSecuredKnownUserVB/LoadStatusToColorConverter.vb
Imports Windows.UI

Public Class LoadStatusToColorConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim statusColor As Color

        Select Case value
            Case Esri.ArcGISRuntime.LoadStatus.Loaded
                statusColor = Colors.Green
            Case Esri.ArcGISRuntime.LoadStatus.Loading
                statusColor = Colors.Gray
            Case Esri.ArcGISRuntime.LoadStatus.FailedToLoad
                statusColor = Colors.Red
            Case Esri.ArcGISRuntime.LoadStatus.NotLoaded
                statusColor = Colors.Red
            Case Else
                statusColor = Colors.Gray
        End Select

        Return New SolidColorBrush(statusColor)
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class

File: src/UWP/ArcGISRuntime.UWP.Viewer/Samples/Security/TokenSecuredKnownUser/LoadStatusToColorConverter.cs
using System;
using Windows.UI;
using Windows.UI.Xaml.Data;
using Windows.UI.Xaml.Media;

namespace TokenSecuredKnownUser
{
    class LoadStatusToColorConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            Color statusColor;

            switch ((int)value)
            {
                case (int)Esri.ArcGISRuntime.LoadStatus.Loaded:
                    statusColor = Colors.Green;
                    break;
                case (int)Esri.ArcGISRuntime.LoadStatus.Loading:
                    statusColor = Colors.Gray;
                    break;
                case (int)Esri.ArcGISRuntime.LoadStatus.FailedToLoad:
                    statusColor = Colors.Red;
                    break;
                case (int)Esri.ArcGISRuntime.LoadStatus.NotLoaded:
                    statusColor = Colors.Red;
                    break;
                default:
                    statusColor = Colors.Gray;
                    break;
            }

            return new SolidColorBrush(statusColor);
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }
}
